selfhost/types.wyv

```diff
@@ -0,0 +1,51 @@
+module types
+
+import wyvern.collections.llist
+type List = llist.LinkedList
+
+resource type Counter
+    var value:Int
+def Counter():Counter = new
+    var value:Int = 0
+
+type Binding
+    val name:String
+    val id:Int
+def Binding(name:String, c:Counter):Binding
+    val count = c.value
+    c.value = c.value + 1
+    new
+        val name = name
+        val id = count
+
+def equalBinding(b1:Binding, b2:Binding):Boolean
+    b1.name == b2.name && b1.id == b2.id
+
+def equalType(t1:Type, t2:Type):Boolean
+    match t1:
+        u:UnitType => match t2:
+            v:UnitType => true
+            default => false
+        n:NominalType => match t2:
+            v:NominalType => n.L == v.L
+            default => false
+
+def stringToType(s:String):Type
+    if (s == "Unit")
+            UnitType()
+        else
+            NominalType(s)
+
+datatype Type
+    UnitType()
+    NominalType(L:String)
```

Comment:
NominalType should contain a Binding, not a String.  Correspondingly, stringToType should take a context and search for the appropriate type binding. This doesn't matter now but it will matter a lot as soon as we support type members.